api/src/main/java/org/openmrs/module/idgen/IdgenModuleActivator.java

```diff
@@ -43,4 +46,18 @@ public void stopped() {
 	public void setDaemonToken(DaemonToken token) {
 		IdgenTask.setDaemonToken(token);
 	}
+	
+	private void initPrefixAttributeType() {
+		for (LocationAttributeType existingType : Context.getLocationService().getAllLocationAttributeTypes()) {
+			if (existingType.getName().equals(IdgenConstants.PREFIX_LOCATION_ATTRIBUTE_TYPE) && !existingType.isRetired()) {
+				// The prefixAttributeType was already created
+				return;
+			}
+		}
+		LocationAttributeType prefixAttributeType = new LocationAttributeType();
+		prefixAttributeType.setName(IdgenConstants.PREFIX_LOCATION_ATTRIBUTE_TYPE);
+		prefixAttributeType.setMinOccurs(0);
```

Comment:
I think that we can assume that each location should be given **maximum** one location prefix, so I believe we should do this:
```java
prefixAttributeType.setMaxOccurs(1);
```
If you don't do this, then you will have to produce unit tests that show that locations with multiple prefixes are handled appropriately.